src/sagas/tokens.js

```diff
@@ -312,29 +348,29 @@ export function* requestUnregisteredTokens(action) {
     return;
   }
 
-  const tokens = {};
-  let someError = false;
-  for (const uid of uids) {
-    try {
-      const { tokenInfo: { symbol, name } } = yield call([wallet, wallet.getTokenDetails], uid);
-      const token = { uid, symbol, name };
-      tokens[uid] = token;
-    } catch (e) {
-      log.error(`Fail getting token data for token ${uid}.`, e);
-      someError = true;
-      // continue download of tokens
+  /**
+   * NOTE: We can improve the follwoing strategy by adopting a more robust
+   * rate-limit algorithm, like the sliding window or token bucket.
+   */
+
+  // These are the default values configured in the nginx conf public nodes.
+  const perBurst = NODE_RATE_LIMIT_CONF.thin_wallet_token.burst;
+  const burstDelay = NODE_RATE_LIMIT_CONF.thin_wallet_token.delay;
+  const uidGroups = splitInGroups(uids, perBurst);
+  for (const group of uidGroups) {
+    // Fork is a non-blocking effect, it doesn't cause the caller suspension.
+    const tasks = yield all(group.map((uid) => fork(getTokenDetails, wallet, uid)));
+    // Awaits a group to finish before burst the next group
+    yield join(tasks);
+    // Skip delay if there is only one group or is the last group
+    if (uidGroups.length === 1 || group === uidGroups.at(-1)) {
+      break;
```

Comment:
I know that technically `break` will work here, as this is the only element in the array, but I think `continue` makes more sense here and will have the same effect

Reply:
Ok. Done.